packages/fuels-signers/src/provider.rs

```diff
@@ -111,9 +116,9 @@ impl Provider {
         &self,
         tx: &Transaction,
     ) -> Result<(TransactionStatus, Vec<Receipt>), ProviderError> {
-        let tx_id = self.client.submit(tx).await?.0.to_string();
+        let tx_id = tx.id().to_string();
+        let status = self.client.submit_and_await_commit(tx).await?;
```

Comment:
Since exeuction is no longer synchronous with tx submission, this new method will poll on the client until the tx has been processed into a block or rejected by the tx pool.